google/cloud/spanner_v1/snapshot.py

```diff
@@ -335,7 +335,7 @@ def read(
                 iterator = _restart_on_unavailable(
                     restart,
                     request,
-                    "CloudSpanner.ReadOnlyTransaction",
+                    f"CloudSpanner.{type(self).__name__}.read",
```

Comment:
Nope, it would be `CloudSpanner.Snapshot.read`; the final class invoking that method is what `type(self)` returns and we have tests as well for this spans :-)